### PR TITLE
Automatically refresh my drains

### PR DESCRIPTION
**Pull request by beyondtheteal**
Contributes to #106 and #132 

### Refresh Drain's info window
1. When a user signs up
2. When a user logs in
3. When a user logs out
4. When one of these actions is done twice in a row. 

### Refresh My Drains list
* When a drain is adopted
* When a drain is abandoned

Note: #4 is broken right now. When you do two of the actions one right after the other, I get the error below
screen shot 2019-03-04 at 10 33 37 pm

### CODE REVIEW

**Comment by Ross-Hunter** [comment]
How would one verify that the info window has refreshed? I'm mucking about and I don't notice anything happen when I sign up.

File: app/assets/javascripts/main.js.erb [hunk 1]
```diff
@@ -271,6 +271,27 @@ $(function() {
       }
     }
   }
+  /* Ajax call to reopen the 'right' info window for the current state */
+  function refreshInfoWindow() {
+    if (activeThingId == null 
+      || activeThingId === null 
+      || activeThingId === undefined) {
```
Comment (Ross-Hunter) [inline app/assets/javascripts/main.js.erb]: I'm so bad at JavaScript that I always rely on lodash for these types of things.

https://lodash.com/docs/4.17.11#isNil

**Comment by beyondtheteal** [comment]
@Ross-Hunter 
The actions that cause the behavior are listed in this comment: https://github.com/citizenlabsgr/adopt-a-drain/issues/154#issue-417070647

File: app/assets/javascripts/main.js.erb (hunk 1 above)
Comment (Wilfongjt) [reply app/assets/javascripts/main.js.erb]: Is lodash dependent on nodeJS?

Comment (Wilfongjt) [reply app/assets/javascripts/main.js.erb]: nodejs is installed in docker but I don't see it in AAD.

**Comment by Ross-Hunter** [comment]
What is the info window and how will I know it changed?

**Comment by Wilfongjt** [comment]
Info windows are a google map thing.  Maybe UI Event is what you are looking for https://developers.google.com/maps/documentation/javascript/events

**Comment by beyondtheteal** [comment]
@Ross-Hunter When you click on a drain, a small window pops up telling you about the drain. There are like 5 different views that can populate in that window, depending on if you are signed in or if it's already adopted, etc.

**Comment by Ross-Hunter** [comment]
My problem is I wasn't viewing an available drain!

This seems to work, though I'm getting 422's randomly and also the email regex being used for validation is bad, it won't let me use `-` in my email.

File: app/assets/javascripts/main.js.erb (hunk 1 above)
Comment (Ross-Hunter) [reply app/assets/javascripts/main.js.erb]: lodash is a front end JS library with no dependencies.

**Comment by beyondtheteal** [comment]
Please let me know if it's required to add lodash for this PR. 
Otherwise, this is ready to be reviewed and merged.

**Pull request merged**